src/routes/level/Level.js

```diff
@@ -89,26 +97,71 @@ class Level extends React.Component {
   };
   render() {
     const { data: { getBestTimes, getLevel, loading } } = this.props;
+    const isWindow = typeof window !== 'undefined';
     return (
-      <div className={s.level}>
-        <Recplayer
-          lev={`/dl/level/${this.props.LevelIndex}`}
-          controls={false}
-        />
-        {loading && <Loading />}
-        {!loading && (
-          <React.Fragment>
-            <h3>{`${getLevel.LevelName}.lev ${getLevel.LongName}`}</h3>
-            <Tabs value={this.state.tab} onChange={this.onTabClick}>
-              <Tab label="Best individual times" />
-              <Tab label="All times" />
-            </Tabs>
-            {this.state.tab === 0 && <TimeTable data={getBestTimes} />}
-            {this.state.tab === 1 && (
-              <AllTimes LevelIndex={this.props.LevelIndex} />
+      <div className={s.root}>
+        <div className={s.playerContainer}>
+          <div className={s.player}>
+            {isWindow && (
+              <Recplayer
+                lev={`/dl/level/${this.props.LevelIndex}`}
+                controls={false}
+              />
             )}
-          </React.Fragment>
-        )}
+          </div>
+        </div>
+        <div className={s.rightBarContainer}>
+          <div className={s.chatContainer}>
+            <ExpansionPanel defaultExpanded>
+              <ExpansionPanelSummary expandIcon={<ExpandMoreIcon />}>
+                <Typography variant="body2">Level info</Typography>
+              </ExpansionPanelSummary>
+              <ExpansionPanelDetails>
```

Comment:
You can either add a loading guard here (as shown on line 146), or if want to show text and stuff even while loading you could just do something like
```javascript
<div className={s.levelDescription}>
  {getLevel ? getLevel.LevelName : ''}.lev
  <div className={s.levelFullName}>
  {getLevel ? getLevel.LongName : ''}
  </div>
```
That would solve the potential undefined error (because getLevel might not be set yet while it's loading data from GraphQL)

https://github.com/elmadev/elmaonline-site/blob/9fa7bce51ba890376c212eb73ce9ddac238e9286/src/routes/level/Level.js#L119-L127

Reply:
Thanks a lot for that! Will implement in the next build :)

Reply:
Fixed with the loading guard, thanks again.